galaxy/scripts/probedb.py

```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/env python3
+
+import argparse
+import psycopg2
+import socket
+import sys
+
+
+from datetime import datetime
+from time import sleep
+
+
+DEBUG=False
+# allowable seconds since db timestamp
+default_interval=60
+
+def dprint(msg):
+    if DEBUG:
+        print(msg)
+
+parser = argparse.ArgumentParser()
+
+
+def add_arg(shortname, longname, default=None, help_text=None, action=None):
+    if len(shortname) != 1:
+        print("ERROR: short flag for argument must be a single character")
+        sys.exit(1)
+    parser.add_argument("-{}".format(shortname), "--{}".format(longname), help=help_text, default=default, action=action)
+
+#    parser.add_argument("-{}".format(shortname), "--{}".format(longname))
+
+args = [
+    { "shortname": "i", "longname": "interval", "help_text": "how long (in seconds) to wait between probes"},
+    { "shortname": "v", "longname": "verbose", "help_text": "enable verbose output", "action": "store_true" },
+    { "shortname": "e", "longname": "environmentvariable", "help_text": "environment variable that contains connection string" },
+    { "shortname": "u", "longname": "username", "help_text": "postgres db user name" },
+    { "shortname": "p", "longname": "password", "help_text": "postgres db user pass" },
+    { "shortname": "d", "longname": "dbname", "help_text": "postgres db name to connect to" },
+    { "shortname": "b", "longname": "dbhost", "help_text": "name of host where postgres lives" },
+    { "shortname": "o", "longname": "hostname", "help_text": "handler host name to be checked in postgres db", "default": socket.gethostname() },
+]
+
+
+def check_rows(connection_string, query_string, interval=60):
+    def less_than_interval(a, b, interval=60):
+        if (b-a).seconds > interval:
+            dprint("time delta {} > {}s".format((b-a).seconds, interval))
+            sys.exit(1)
+        dprint("time delta {} <= {}s".format((b-a).seconds, interval))
+        sys.exit(0)
+
+    conn = psycopg2.connect(connection_string)
+    cur = conn.cursor()
+    cur.execute(query_string)
+    rows = cur.fetchall()
+    if len(rows) != 1:
+        dprint("ERROR: there should be only one row, but {} found".format(len(rows)))
+        exit(1)
+    for row in rows:
+        now = datetime.now()
+        less_than_interval(row[1], now, interval)
+
+
+for arg in args:
+    add_arg(**arg)
```

Comment:
It's usually more common to define parser args as follows:
https://github.com/galaxyproject/galaxy/blob/6da7b1ade8866dd9048881c20e37815a79e50f89/scripts/cleanup_datasets/cleanup_datasets.py#L76
This is because parser args are often not this uniform, and it's common to have additional parameters.

Reply:
suggested change [implemented](https://github.com/luke-c-sargent/galaxy-helm/blob/10e20ed149970e0ab6bdca95192726cca26b4262/galaxy/scripts/probedb.py#L44-L62)